FWCore/Services/plugins/CPU.cc

```diff
@@ -30,18 +31,21 @@
 namespace edm {
   
   namespace service {
-    class CPU {
+    class CPU : public CPUServiceBase {
     public:
       CPU(ParameterSet const&, ActivityRegistry&);
       ~CPU();
       
       static void fillDescriptions(ConfigurationDescriptions& descriptions);
-      
+
+      virtual bool cpuInfo(std::string &models, double &avgSpeed) override;
+
     private:
       int totalNumberCPUs_;
       double averageCoreSpeed_;
       bool reportCPUProperties_;
-      
```

Comment:
Updating the member data in the cpuInfo call is not thread safe. As far as I can tell, there is no need for `totalNumberCPUs_` or `averageCoreSpeed_` to be member data. It also looks like `reportCPUProperties_` could be `const`.